Updated Heatmiser climate requirements and allow sensor choice

bretttickx opened the pull request:
## Description:

- fixed requirement using latest heatmiserV3 
- sensor choice between floor and air
- can now turn thermostat to frost mode and alter min max temp
- fixed precision now whole degrees
- improved polling by force first thermostat in list to call connection

**Related issue (if applicable):** fixes #13146

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#<home-assistant.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml

climate:
  - platform: heatmiser
    ipaddress: 192.168.1.87
    port: 26
    tstats:
      - name: Living Room
        id: 1
        model: prt
        sensor: air
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23

bretttickx committed:
Update climate.py

bretttickx committed:
Updated component requirements and allow sensor choice

- fixed requirement using latest heatmiserV3 
- sensor choice between floor and air
- can now turn thermostat to frost mode and alter min max temp
- fixed precision now whole degrees
- improved polling by force first thermostat in list to call connection

bretttickx committed:
removed commented out code

bretttickx committed:
Merge pull request #1 from bretttickx/hotfix-heatmiser

Updated component requirements and allow sensor choice

homeassistant commented:
Hi @bretttickx, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#22448).

Once you do that we will be able to review and accept this pull request.

Thanks!

cgarwood commented on homeassistant/components/heatmiser/climate.py:
```diff
@@ -5,72 +5,81 @@
 _LOGGER = logging.getLogger(__name__)
 
 CONF_IPADDRESS = 'ipaddress'
+CONF_PORT = 'port'
```
`from homeassistant.const import CONF_PORT`

bretttickx replied:
thanks for your help :-) not sure what i need to do here as I'm a python noob! haha I've just been hacking to try fix the original component is it because theirs a conflict in variable names with the home assistant constants variables??

cgarwood replied:
`CONF_PORT` is already defined in the main home assistant const file, so we should just import from there instead of redefining it.

cgarwood commented on homeassistant/components/heatmiser/climate.py:
```diff
@@ -5,72 +5,81 @@
-
+            thermostat, room, sensor, uh1_per, def_min_temp, def_max_temp)
+        ])
+        time.sleep(2.4)
```
Using sleep() is usually a big no-no since it's blocking. What's the purpose of this sleep call?

bretttickx replied:
the heatmiser thermostats uses rs485 serial connection , and needs a 3 second delay between comms, the original component had this delay i've just tweaked it down as far as i can before the performance starts to suffer.

bretttickx committed:
fix formatting issues and small chnages

- fixed houndci formatting issues
- fixed issue with clashing HA constant variable

bretttickx committed:
houdci formatting issues

bretttickx committed:
more hound-ci issues

bretttickx committed:
more hounding!

bretttickx commented:
Hi @cgarwood,  I'm I missing something to go forward or do the travis ci builds just take a long time to process?

bretttickx committed:
- more code markup changes travis build fails

bretttickx committed:
- added requirements all?
- more linting issues

bretttickx committed:
- added blank line at end of file arrgh

bretttickx committed:
Update climate.py

- update to new component structure

bretttickx committed:
added manifest file

bretttickx committed:
Merge branch 'dev' into dev

bretttickx committed:
derps

bretttickx committed:
derps

bretttickx committed:
derps PEP8

bretttickx committed:
- update requirements all

balloob commented:
We're redoing climate architecture in #23899 . All PRs touching climate related code are blocked until that is done.

bretttickx commented:
> We're redoing climate architecture in #23899 . All PRs touching climate related code are blocked until that is done.

@balloob Ok Cool, will wait till this done hopefully i can get it over the line as I've had it working for some time as a custom component be nice to see it in the repo

MartinHjelmare commented:
The climate remake is now done. Please update this PR according to the new architecture:
https://developers.home-assistant.io/blog/2019/07/03/climate-cleanup.html

MartinHjelmare commented:
There hasn't been any movement here for some time now. I'll close now. We're trying to decrease our open PR buffer. Please open a new PR when ready to finish. Thanks!

bennealon commented:
Hi guys, has anyone got appetite to look at this? I'm having issues with trying to connect to my heatmeiser neostat's, im getting a CRC failure and wondered if this might help fix that?

MartinHjelmare commented:
Please open an issue if you suspect a bug.

If you want to suggest an enhancement please open a feature request in [the Feature Requests section](https://community.home-assistant.io/c/feature-requests) of our community forum.

Closed PRs should not be used for enhancement discussion or bug reports.

Thanks!
